code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/InAppSchemaData.java

```diff
@@ -0,0 +1,110 @@
+/*
+  Copyright 2023 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+ */
+
+package com.adobe.marketing.mobile.messaging;
+
+import static com.adobe.marketing.mobile.messaging.MessagingConstants.ConsequenceDetailDataKeys.CONTENT;
+import static com.adobe.marketing.mobile.messaging.MessagingConstants.ConsequenceDetailDataKeys.CONTENT_TYPE;
+import static com.adobe.marketing.mobile.messaging.MessagingConstants.ConsequenceDetailDataKeys.EXPIRY_DATE;
+import static com.adobe.marketing.mobile.messaging.MessagingConstants.ConsequenceDetailDataKeys.METADATA;
+import static com.adobe.marketing.mobile.messaging.MessagingConstants.ConsequenceDetailDataKeys.MOBILE_PARAMETERS;
+import static com.adobe.marketing.mobile.messaging.MessagingConstants.ConsequenceDetailDataKeys.PUBLISHED_DATE;
+import static com.adobe.marketing.mobile.messaging.MessagingConstants.ConsequenceDetailDataKeys.REMOTE_ASSETS;
+import static com.adobe.marketing.mobile.messaging.MessagingConstants.ConsequenceDetailDataKeys.WEB_PARAMETERS;
+
+import com.adobe.marketing.mobile.services.Log;
+import com.adobe.marketing.mobile.util.JSONUtils;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+// represents the schema data object for an in-app schema
+public class InAppSchemaData {
+    private static final String LOG_TAG = "Messaging";
+    private static final String SELF_TAG = "InAppSchemaData";
+    private Object content;
+    private ContentType contentType;
+    private int publishedDate;
+    private int expiryDate;
+    private Map<String, Object> meta;
+    private Map<String, Object> mobileParameters;
+    private Map<String, Object> webParameters;
+    private List<String> remoteAssets = new ArrayList<>();
+
+    InAppSchemaData(final JSONObject schemaData) {
+        try {
+            this.contentType = ContentType.fromString(schemaData.optString(CONTENT_TYPE));
+            if (contentType.equals(ContentType.APPLICATION_JSON)) {
+                this.content = JSONUtils.toMap(schemaData.getJSONObject(CONTENT));
```

Comment:
Same question here. Can this be a `JSONArray`? More importantly this cannot be anything other than HTML considering it is constructed by AJO. Should we allow support for other content types?

Reply:
like feed items, in-app are JSON objects.